src/pages/permissions/CreateTLSIdentity.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState, type FC } from "react";
+import CreateIdentityModal from "./CreateIdentityModal";
+import CreateTLSIdentityPanel from "./panels/CreateTLSIdentityPanel";
+import { usePortal } from "@canonical/react-components";
+import usePanelParams, { panels } from "util/usePanelParams";
+
+const CreateTLSIdentity: FC = () => {
+  const panelParams = usePanelParams();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const [token, setToken] = useState("");
+  const [error, setError] = useState(false);
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <CreateIdentityModal
+            onClose={closePortal}
+            token={token}
+            error={error}
+          />
+        </Portal>
+      )}
+
+      {panelParams.panel === panels.createTLSIdentity && (
+        <CreateTLSIdentityPanel
+          openPortal={openPortal}
+          setToken={setToken}
+          setError={setError}
```

Comment:
With moving the error handling into the panel, the interface can slightly simplify to decouple the two components:
```suggestion
          onSuccess={setToken}
```